plugins/parsers/registry.go

```diff
@@ -222,7 +226,11 @@ func NewParser(config *Config) (Parser, error) {
 		parser, err = NewValueParser(config.MetricName,
 			config.DataType, config.ValueFieldName, config.DefaultTags)
 	case "influx":
-		parser, err = NewInfluxParser()
+		if config.InfluxParserVersion == 2 {
+			parser, err = NewInfluxParserv2()
+		} else {
+			parser, err = NewInfluxParser()
+		}
```

Comment:
@reimda was this what you were thinking of with the different parser types?

As written, the `NewInfluxParser` remains the default unless 2 is explicitly called out. In a future release, we can swap this to only use v1 if explicitly called out, and then finally we can go back to only using v2?

Right now, I have the code separate as `influx` and `influx_v2` parsers. The user doesn't know or need to care about the naming of these. I found this simpler than trying to do lots of if/else statements or further complicating the existing parser code. Thoughts on that?

Please feel free to bikeshed the naming, I could not think of anything quickly.

Reply:
Having a setting like this is just what I was thinking.

About naming, "v2" has been overloaded in a few ways already and I think it's getting confusing for users. There's v1 and v2 of influxdb itself, telegraf has influxdb_listener and influxdb_listener_v2 input plugins, and influxdb and influxdb_v2 output plugins.

I wonder if naming this setting "version" and having people use the new parser by setting it to 2 is going to mislead or confuse people. I could see people thinking they need to use parser version 2 if they're using influxdb_listener_v2 or influxdb 2. The new parser needs to work with all these just like the old parser does.

Since the new parser isn't really the next version of telegraf's current parser, but a separate parser in a separate go module, maybe we should have it be a string settting that accepts "internal" for the existing parser or "module" for the new parser. Maybe the name of the setting is just "influx_parser".

Reply:
Upstream and internal sound good to me